src/robot_arm.cpp

```diff
@@ -1,19 +1,69 @@
+/**
+ * @file
+ * @brief     Robot Arm class
+ * @author    Jeroen van Hattem and Jeffrey de Waal
+ * @license   MIT
+ */
 #include "robot_arm.hpp"
 
-void robot_arm::move(int coordinates[3]) {
-    copy_data(coordinates, goto_coordinates);
+RobotArm::RobotArm() : conn(115200, UARTController::ONE) {
+}
+
+void RobotArm::move(int coordinates[3], int _speed) {
+    for (unsigned int i = 0; i < 3; i++) {
+        goto_coordinates[i] = coordinates[i];
+    }
+
+    if ((hwlib::now_us() / 1000) - startMsSend > 2500) {
+        startMsSend = hwlib::now_us() / 1000;
+        conn << "G0 X150 Y150 Z150 F10000\n"; /// Move
+    }
 
-    hwlib::cout << "Moving to (" << coordinates[0] << ", " << coordinates[1] << ", " << coordinates[2] << ")" << hwlib::endl;
-    output_pin.set(1);
-    hwlib::wait_ms(1000);
-    output_pin.set(0);
-    hwlib::cout << "Finished moving" << hwlib::endl;
+    if (conn.available() > 0 && (hwlib::now_us() / 1000) - startMsReceive > 30) {
+        startMsReceive = hwlib::now_us() / 1000;
 
-    copy_data(goto_coordinates, current_coordinates);
+        for (unsigned int i = 0; i < conn.available(); i++) {
+            hwlib::cout << (char)conn.receive();
+        }
+    }
+
+    for (unsigned int i = 0; i < 3; i++) {
+        current_coordinates[i] = goto_coordinates[i];
+    }
+    speed = _speed;
 }
 
-void robot_arm::copy_data(int *array1, int *array2) {
-    for (int i = 0; i < (sizeof(array1) / 4); i++) {
-        array2[i] = array1[i];
+void RobotArm::executeAction(Actions action) {
+    action = action;
+    hwlib::cout << "Executing Action" << hwlib::endl;
+}
+
+void RobotArm::determineGCode(int coordinates[3]) {
+    hwlib::cout << "Moving to " << coordinates[0] << ", " << coordinates[1] << ", " << coordinates[2] << "." << hwlib::endl;
+}
+
+void RobotArm::determineGCode(Actions action) {
+    switch (action) {
```

Comment:
Why do you use a switch? Also, why do you not break the cases? If this is on purpose, document it.